src/mmocore/EventEmitter.ts

```diff
@@ -58,23 +58,30 @@ export default class EventEmitter {
     }
   }
 
+  onAll(handler: EventHandler): boolean {
+    return this.on("*", handler);
+  }
+
   offAll(): void {
     this._eventHandlers = {};
   }
 
   fire(type: string, data?: Record<string, unknown>): void {
     if (!type) return;
 
-    const handlers = this._eventHandlers[type];
-    if (!handlers || !handlers.length) return;
+    let handlers = this._eventHandlers[type] ?? [];
+    const allHandlers = this._eventHandlers["*"] ?? [];
+    if (allHandlers && allHandlers.length)
+      handlers = handlers.concat(allHandlers);
+    if (!handlers.length) return;
 
     const event = this.createEvent(type, data);
 
     for (const handler of handlers) {
-      if (handler._once) event.once = true;
+      const once = (event.once = handler._once === true);
 
       handler(event);
-      if (event.once) this.off(type, handler);
+      if (once) setImmediate(this.off.bind(this), type, handler); // we should not modify handlers in for loop
```

Comment:
do we need the async _setImmediate_, if so, lets use setTimeout ?

Reply:
This is the exact place where it's needed the most, I can stick to `setTimeout` although setImmediate is commonly used with node.js.